### PR TITLE
make clean-clusters is deleting images that it probably shouldn't

### LINKED ISSUE #698: `make clean-clusters` is deleting docker images that it probably shouldn't
Running `make deploy` and `make clean-clusters` multiple times from a server on corporate network and started hitting the Docker pull rate issue:
```
docker: Error response from daemon: toomanyrequests: You have reached your pull rate limit.
  You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit. 
```
Looks like on `make clean-clusters`, the following images are being deleted and probably shouldn't be:
```
$ docker images
REPOSITORY                                             TAG               IMAGE ID       CREATED              SIZE
:
localhost:5000/submariner-operator-index               <none>            7de88ff5f803   About an hour ago    37MB
:
kindest/node                                           <none>            14c4b042989b   7 months ago         1.08GB
:
```

### PR DESCRIPTION
**Pull request by Billy99**
Running make deploy and make clean-clusters multiple times from a server on
corporate network and started hitting the Docker pull rate issue:
  "toomanyrequests: You have reached your pull rate limit"

Preload KIND image so docker pull failure can be added to a output signalling
a manual pull required. Also tag the image so that it is not considered a
"dangling" image and deleted by "docker system prune" on "make clean-clusters".

Closes: #698

Signed-off-by: Billy McFall <22157057+Billy99@users.noreply.github.com>

### CODE REVIEW

File: scripts/shared/clusters.sh [hunk 1]
```diff
@@ -69,6 +69,41 @@ function kind_fixup_config() {
...
 }
 
+function download_kind() {
+    # In development environments where clusters are brought up and down
+    # multiple times, several Docker images are repeatedly pulled and deleted,
+    # leading to the Docker error:
+    #   "toomanyrequests: You have reached your pull rate limit"
+    # Preload the KIND image. Also tag it so the `docker system prune` during
+    # cleanup won't remove it.
+    if [[ -n ${k8s_version} ]]; then
```
Comment (mkolesnik) [inline scripts/shared/clusters.sh]:
```suggestion
    if [[ -z "${k8s_version}" ]]; then
        echo "k8s_version not set."
        return
    fi
```

File: scripts/shared/clusters.sh [hunk 2]
```diff
@@ -69,6 +69,41 @@ function kind_fixup_config() {
     chmod a+r $KUBECONFIG
 }
 
+function download_kind() {
+    # In development environments where clusters are brought up and down
+    # multiple times, several Docker images are repeatedly pulled and deleted,
+    # leading to the Docker error:
+    #   "toomanyrequests: You have reached your pull rate limit"
+    # Preload the KIND image. Also tag it so the `docker system prune` during
+    # cleanup won't remove it.
```
Comment (mkolesnik) [inline scripts/shared/clusters.sh]:
```suggestion
# In development environments where clusters are brought up and down
# multiple times, several Docker images are repeatedly pulled and deleted,
# leading to the Docker error:
#   "toomanyrequests: You have reached your pull rate limit"
# Preload the KIND image. Also tag it so the `docker system prune` during
# cleanup won't remove it.
function download_kind() {
```

Wouldn't this make more sense?

**Review by dfarrell07 — APPROVED** [review]
LGTM, thanks @Billy99 and sorry for the slow review!

**Comment by dfarrell07** [comment]
> /retest

We don't have Prow here, you have to click into the failing GHA and click "Rerun all jobs" or similar.

**Review by sridhargaddam — APPROVED** [review]
Thanks for the PR @Billy99

File: scripts/shared/clusters.sh [hunk 3]
```diff
@@ -69,6 +69,41 @@ function kind_fixup_config() {
...
+        # Example: kindest/node:v1.20.7@sha256:cbeaf907fc78ac97ce7b625e4bf0de16e3ea725daf6b04f930bd14c67c671ff9
+        kind_image="kindest/node:v${k8s_version}"
+        # Example: kindest/node:v1.20.7
+        kind_image_tag="kindest/node:v$(echo ${k8s_version}|awk -F"@" '{print $1}')"
+        # Example: kindest/node:@sha256:cbeaf907fc78ac97ce7b625e4bf0de16e3ea725daf6b04f930bd14c67c671ff9
+        kind_image_sha="kindest/node@$(echo ${k8s_version}|awk -F"@" '{print $2}')"
+
+        # Check if image is already present, and if not, download it.
+        echo "Processing Image: $kind_image_tag ($kind_image)"
+        if [[ -z $(docker images -q "$kind_image_tag") ]] ; then
```
Comment (mkolesnik) [inline scripts/shared/clusters.sh]:
```suggestion
        if [[ -n $(docker images -q "$kind_image_tag") ]] ; then
            echo "Image $kind_image_tag already downloaded."
            return
        fi
```

**Review by mkolesnik — CHANGES_REQUESTED** [review]
Let's have less nesting to make it easier to read and follow

File: scripts/shared/clusters.sh (hunk 2 above)
Comment (Billy99) [reply scripts/shared/clusters.sh]: Done

File: scripts/shared/clusters.sh (hunk 1 above)
Comment (Billy99) [reply scripts/shared/clusters.sh]: Done

File: scripts/shared/clusters.sh (hunk 3 above)
Comment (Billy99) [reply scripts/shared/clusters.sh]: Done

**Review by Billy99 — COMMENTED** [review]
Since they were relatively minor changes, I collapsed the review comment changes instead of adding as separate commit.

**Review by skitt — APPROVED** [review]

**Comment by nyechiel** [comment]
@mkolesnik can you re-review please?

**Comment by nyechiel** [comment]
@mkolesnik let's get this one in

**Review by mkolesnik — APPROVED** [review]

**Pull request merged**